### PR TITLE
[feat][fix] delta neutral vault set config and oracle

### PR DESCRIPTION
**Pull request by jr-alpaca**
## Description

- change maxVaultPositionValue to public modifier
- add setPriceOracle and setDeltaVaultConfig to DeltaNeutralVault
## ChangeLogs:

- DeltaNeutralVault.sol
- DeltaNeutralVaultConfig.sol

### CODE REVIEW

**Review by alpanaca — APPROVED** [review]

**Comment by alpanaca** [comment]
LGTM :thumbsup:

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 1]
```diff
@@ -525,6 +525,57 @@ describe("DeltaNeutralVault", () => {
       });
     });
   });
+
+  describe("#setConfig", async () => {
```
Comment (abc-alpaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:529]: do we need when non owner set new config?

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 2]
```diff
@@ -697,6 +699,20 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
     return _totalPositionValue - _totalDebtValue;
   }
 
+  /// @notice Set new DeltaNeutralOracle.
+  /// @param _newPriceOracle New deltaNeutralOracle address.
+  function setDeltaNeutralOracle(IDeltaNeutralOracle _newPriceOracle) external onlyOwner {
+    priceOracle = _newPriceOracle;
```
Comment (spicysquid168) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:705]: please add sanity call to _newPriceOracle to make sure it is really IDeltaNeutralOracle

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 3]
```diff
@@ -697,6 +699,20 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
+    emit LogSetDeltaNeutralOracle(msg.sender, address(_newPriceOracle));
+  }
+
+  /// @notice Set new DeltaNeutralVaultConfig.
+  /// @param _newVaultConfig New deltaNeutralOracle address.
+  function setDeltaNeutralVaultConfig(IDeltaNeutralVaultConfig _newVaultConfig) external onlyOwner {
+    config = _newVaultConfig;
```
Comment (spicysquid168) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:712]: same as above

**Comment by spicysquid168** [comment]
2 comments above

**Comment by jr-alpaca** [comment]
fixed all comments @spicysquid168 @abcdefu-alpaca

**Review by abc-alpaca — APPROVED** [review]

**Comment by abc-alpaca** [comment]
approved

**Review by spicysquid168 — APPROVED** [review]

**Comment by spicysquid168** [comment]
approved

**Pull request merged**